### PR TITLE
activation response parsing + success submitted changes

### PR DESCRIPTION
**Pull request by dougperez**

### CODE REVIEW

File: lib/conduit/thingspace/parsers/base.rb [hunk 1]
```diff
@@ -32,7 +33,17 @@ def initialize(json)
...
       def response_status
-        response_errors.any? ? "failure" : "submitted"
+        if response_errors.any?
+          "failure"
+        elsif success_response?
+          "success"
+        else
+          "submitted"
+        end
+      end
```
Comment (ufchrisg) [inline lib/conduit/thingspace/parsers/base.rb]: this should mean we no longer need the overrides in `start_session`, `get_oauth`, and `get_device_info`

File: lib/conduit/thingspace/request_mocker/fixtures/responses/activate/success.json.erb [hunk 2]
```diff
@@ -1,3 +1 @@
-{
-  "requestId": "595f5c44-c31c-4552-8670-020a1545a84d"
-}
+{"username":"testuser","password":"testpass","requestId":"41b48101-e09d-414f-9b0e-e95a917b2050","deviceIds":[{"id":"358046082269999","kind":"IMEI"},{"id":"89148000003793769999","kind":"ICCID"}],"deviceResponse":{"activateResponse":{"deviceIds":[{"id":"5612849999","kind":"mdn"},{"id":"311480385429999","kind":"imsi"},{"id":"358046082269999","kind":"imei"},{"id":"15612849999","kind":"msisdn"},{"id":"89148000003793769999","kind":"iccid"},{"id":"5615129999","kind":"min"}],"state":"Active","servicePlan":"PLANA","featureCodes":[],"deviceCredential":{"username":"5612849999@vzw3g.com","password":"5612849999"}}},"status":"Success","callbackCount":1,"maxCallbackThreshold":4}
```
Comment (ufchrisg) [inline lib/conduit/thingspace/request_mocker/fixtures/responses/activate/success.json.erb]: why does this one not get to look pretty? :(

File: lib/conduit/thingspace/parsers/base.rb (hunk 1 above)
Comment (pedromartinez) [reply lib/conduit/thingspace/parsers/base.rb]: great

File: lib/conduit/thingspace/request_mocker/fixtures/responses/activate/success.json.erb (hunk 2 above)
Comment (dougperez) [reply lib/conduit/thingspace/request_mocker/fixtures/responses/activate/success.json.erb]: Made pretty.

**Comment by pedromartinez** [comment]
This looks great. One note. We need to handle async responses that result in a failure. Example:

image

**Comment by dougperez** [comment]
@pedromartinez added failure handling.

**Comment by pedromartinez** [comment]
Well done sir

**Pull request merged**